### PR TITLE
Upgrade ThreeJS and use WebGL 2

### LINKED ISSUE #5745: Consider using WebGL 2
## Detailed Description
Since iOS 15 supports webgl 2, we can consider switching to it. First, we should find out the impact of this (performance gains, code simplification?). Second, we should estimate the effort (including testing for iPad).

### PR DESCRIPTION
**Pull request by philippotto**
### Summary
- Enable WebGL2
- Update ThreeJS to r137 to fix issues with RG textures
- Replace GLSL validation package to support webgl 2
- Update node to 16 since new glsl package requires this
- Remove shader rendering spec due to incompatibilities with node 16 (see https://app.circleci.com/pipelines/github/scalableminds/webknossos/7286/workflows/e17826f1-461a-481c-ba60-246174b32df5/jobs/18622/parallel-runs/0/steps/0-109). there was only one spec, anyway, with very low impact. the e2e screenshot tests serve us better here.

The screenshots have slightly thicker borders (the red, green, blue ones) now apparently:
old: 
image

new: 
image

I'm not sure whether this is only in the screenshot tests or for the normal browser setup, too. However, I don't think that this is a blocker. Especially, because the pixel difference is so small that the screenshot tests pass.

### URL of deployed dev instance (used for testing):
- https://webgl.webknossos.xyz

### Todo:
- [x] clarify what cross sections are and check whether they work? (see cube.ts)

### Steps to test:
- [X] test on iPad (I already did this via browserstack and the rendering seems to work)
- [X] run nightly screenshot test against this branch
Also, try to test wk rather broad regarding rendering, but the following were changed during this PR which is why they should be tested with extra care:
- crosshair
	- in ortho
	- in flight mode
- mesh smoothing / computeVertexNormals
- Quaternion usage in 3D viewport / use the rotate buttons
- rendering of "cubes"

### Issues:
- fixes #5745 

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/controller/scene_controller.ts
Comment (philippotto) [inline frontend/javascripts/oxalis/controller/scene_controller.ts:235]: is there a noticeable setback due to this change?

File: frontend/javascripts/oxalis/geometries/cube.ts
Comment (philippotto) [inline frontend/javascripts/oxalis/geometries/cube.ts:105]: todo

Comment (philippotto) [reply frontend/javascripts/oxalis/geometries/cube.ts:105]: @daniel-wer you probably have some insight here :)

Comment (daniel-wer) [reply frontend/javascripts/oxalis/geometries/cube.ts:105]: As far as I know, the cross sections simulate that the bounding box is a cuboid although it is actually only a wireframe of a cuboid. This is important so that the bounding box can be seen whenever one of its *faces* is visible in one of the viewports and not only whenever one of its *edges* is visible.
This is done by having one cross section per viewport which is a simple rectangle with 4 edges and moving it in the respective 3rd dimension whenever the flycam position changes.
Currently, this doesn't seem to work on the deployed branch.
Hope this makes sense, otherwise let's have a quick call :)

**Comment by daniel-wer** [comment]
Quick note because I encountered this during a test:
- [ ] Downloading a computed ad-hoc mesh from the Segments list fails with ` TypeError: Da.Geometry is not a constructor`

File: frontend/javascripts/oxalis/geometries/cube.ts
Comment (philippotto) [reply frontend/javascripts/oxalis/geometries/cube.ts:105]: thank you, makes total sense :+1:

File: frontend/javascripts/libs/UpdatableTexture.ts
Comment (daniel-wer) [inline frontend/javascripts/libs/UpdatableTexture.ts:4]: Good job, cleaning this up and making it type-check :1st_place_medal:

**Review by daniel-wer — COMMENTED** [review]
Awesome, great milestone, thank you for your persistence in getting this to work! Code mostly LGTM, I'll do some proper testing and will report back, afterwards :)

File: frontend/javascripts/oxalis/geometries/cube.ts
Comment (philippotto) [reply frontend/javascripts/oxalis/geometries/cube.ts]: should be fixed now :tada:

File: frontend/javascripts/oxalis/model/bucket_data_handling/texture_bucket_manager.ts
Comment (philippotto) [reply frontend/javascripts/oxalis/model/bucket_data_handling/texture_bucket_manager.ts:39]: it was wrong (outdated) before :)

File: frontend/javascripts/oxalis/controller/renderer.ts
Comment (philippotto) [reply frontend/javascripts/oxalis/controller/renderer.ts:16]: no, it's meant to clarify that `WebGLRenderer` indeed refers to a webgl2 renderer :see_no_evil:

**Comment by daniel-wer** [comment]
Notes from testing:
- [ ] The connectome_file_test_dataset prediction layer (uint24!) seems to have gaps in z, see screenshot of YZ viewport (shouldn't this have been catched by the screenshot tests? :thinking:)
  connectome_file_test_dataset_gaps
- [ ] Subjectively, I encounter missing buckets in the viewports much more often than usual, sometimes on page load or when moving. Could also be that the server is under heavy load :shrug: (Actually, I encountered this on master as well, so the culprit seems to be the [nine_float_layer_dataset](https://master.webknossos.xyz/datasets/sample_organization/nine_float_layer_dataset/view) or the dev server, not an issue of this PR)
  missing_buckets
  missing_buckets_2

File: frontend/javascripts/oxalis/controller/renderer.ts
Comment (daniel-wer) [reply frontend/javascripts/oxalis/controller/renderer.ts:16]: Ah makes sense, I think that's fine :+1:

**Comment by philippotto** [comment]
> The brush preview sphere seems to be much brighter than usual. Ist this by design?

No, this was not intentional. I think some default behavior regarding the alpha value changed. I pushed a fix, which improves the color. I think, it's not 100% the same as on master, but it should be fine (and it's probably not worth the hassle to achieve the exact same output as before).

>  The connectome_file_test_dataset prediction layer (uint24!) seems to have gaps in z, see screenshot of YZ viewport (shouldn't this have been catched by the screenshot tests? :thinking: )

good catch :+1: The screenshot test for uint24 seems to test a quite unfortunate position (see image below). I'll look into this.

image

**Comment by philippotto** [comment]
@daniel-wer I think, the PR is ready for a second look. I fixed all the issues that are relevant to this PR and the screenshot tests should be good now, too. I'll probably have our team do some additional testing, too, to get a good coverage of different systems (hardware, OS, browser etc).

File: .circleci/not-on-master.sh
Comment (daniel-wer) [inline .circleci/not-on-master.sh:4]: Reminder to re-enable this before merging

File: frontend/javascripts/oxalis/geometries/cube.ts
Comment (daniel-wer) [inline frontend/javascripts/oxalis/geometries/cube.ts:138]: Is this really needed, here and if so why only for the XZ plane?

**Review by daniel-wer — APPROVED** [review]
Nice fixes, everything:tm: working as intended now :+1:

File: frontend/javascripts/oxalis/geometries/cube.ts
Comment (philippotto) [reply frontend/javascripts/oxalis/geometries/cube.ts]: good catch. this wasn't intentional, as this is already done by `updatePositionForCrossSections` for all planes.

**Pull request merged**
